### PR TITLE
[tubitv] TV series extraction fix & metadata improvements

### PR DESCRIPTION
**Pull request by bbepis**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fixes tubitv's TV series extraction (TubiTvShowIE) by handling `new Date("..")` values in embedded JS blobs. Also adds additional metadata to the main extractor when the title matches a series : episode format

### CODE REVIEW

File: yt_dlp/extractor/tubitv.py [hunk 1]
```diff
@@ -119,6 +119,15 @@ def _real_extract(self, url):
             'release_year': int_or_none(video_data.get('year')),
         }
 
+        episode_match = re.search(r'^S(\d+):E(\d+) - (.+)', title)
+
+        if episode_match:
+            infodict['season_number'] = int(episode_match.group(1))
+            infodict['episode_number'] = int(episode_match.group(2))
```
Comment (Ashish0804) [inline yt_dlp/extractor/tubitv.py:126]: use int_or_none

Comment (pukkandan) [reply yt_dlp/extractor/tubitv.py:126]: Since the match is done with `\d+`, int is fine

File: yt_dlp/extractor/tubitv.py [hunk 2]
```diff
@@ -119,6 +119,15 @@ def _real_extract(self, url):
...
+            infodict['episode_title'] = episode_match.group(3)
+
+        return infodict
```
Comment (pukkandan) [inline yt_dlp/extractor/tubitv.py:129]:
```py
        season_number, episode_number, episode_title = self._search_regex(
            r'^S(\d+):E(\d+) - (.+)', ..., groups=(1, 2, 3), default=(None, None, None))
        return {
            ...
            'episode': episode_title,
            ...
        }
```

File: yt_dlp/extractor/tubitv.py [hunk 3]
```diff
@@ -132,9 +141,13 @@ class TubiTvShowIE(InfoExtractor):
...
-        show_json = self._parse_json(self._search_regex(
-            r"window\.__data\s*=\s*({.+?});\s*</script>",
-            show_webpage, 'data',), show_name, transform_source=js_to_json)['video']
+
+        raw_json = self._search_regex(
+            r"window\.__data\s*=\s*({[^<]+});\s*</script>",
+            show_webpage, 'data')
+
+        show_json = self._parse_json(raw_json, show_name, transform_source=js_to_json)['video']
+
```
Comment (pukkandan) [inline yt_dlp/extractor/tubitv.py:150]: unnecessary changes. Only the regex needs to be changed
```suggestion
        show_json = self._parse_json(self._search_regex(
            r'window\.__data\s*=\s*({[^<]+});\s*</script>',
            show_webpage, 'data'), show_name, transform_source=js_to_json)['video']
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Comment by Jules-A** [comment]
Sorry this is unrelated to this PR but I wasn't able to get an answer before. Is "_GEO_COUNTRIES = ['US']" needed? It doesn't allow for content outside of my country for me.

File: yt_dlp/extractor/tubitv.py (hunk 3 above)
Comment (bbepis) [reply yt_dlp/extractor/tubitv.py]: Done

File: yt_dlp/extractor/tubitv.py (hunk 2 above)
Comment (bbepis) [reply yt_dlp/extractor/tubitv.py]: Didn't know about this syntax. Very cool, added it

**Comment by bbepis** [comment]
@Jules-A I'm not sure myself, since I never used the geo-bypass functionality because I never needed to use it.

I believe it only affects the extractor if you have the geo bypass flag set?

**Comment by pukkandan** [comment]
@Jules-A issue number? I must have missed it.

It allows users from other countries to access the US content. For content not available in US, you can workaround by adding --no-geo-bypass. Ideally, we should be able to automatically detect whether you are already in a supported region. But can't say whether it is possible without more details

**Comment by Jules-A** [comment]
> @Jules-A issue number? I must have missed it.
> 
> It allows users from other countries to access the US content. For content not available in US, you can workaround by adding --no-geo-bypass or by setting --geo-bypass-country manually. Ideally, we should be able to automatically detect whether you are already in a supported region. But can't say whether it is possible without more details

I didn't create a new issue but I've posted on another semi-relevant issue and on Discord but I couldn't get an answer on if it worked for anyone. It doesn't work for me from Australia, maybe it did in the past but it's possible they've put better restrictions in place?

**Comment by pukkandan** [comment]
Please open an issue. This conversation will just get lost here
